client/web/src/search/results/streaming/StreamingSearchResults.test.tsx

```diff
@@ -0,0 +1,214 @@
+
+    const streamingSearchResult: AggregateStreamingSearchResults = {
+        results: MULTIPLE_SEARCH_RESULT.results,
+        filters: MULTIPLE_SEARCH_RESULT.dynamicFilters,
+        progress: {
+            done: true,
+            durationMs: 500,
+            matchCount: MULTIPLE_SEARCH_RESULT.matchCount,
+            skipped: [],
+        },
+    }
+
+    const defaultProps: StreamingSearchResultsProps = {
+        caseSensitive: false,
+        setCaseSensitivity: sinon.spy(),
+        patternType: SearchPatternType.literal,
+        setPatternType: sinon.spy(),
+        versionContext: undefined,
+
+        extensionsController,
+        telemetryService: NOOP_TELEMETRY_SERVICE,
+
+        history,
+        location: history.location,
+        authenticatedUser: null,
+
+        navbarSearchQueryState: { query: '', cursorPosition: 0 },
+
+        settingsCascade: {
+            subjects: null,
+            final: null,
+        },
+        platformContext: { forceUpdateTooltip: sinon.spy(), settings: NEVER },
+
+        streamSearch: () => of(streamingSearchResult),
+    }
+
+    it('should call streaming search API with the right parameters from URL', () => {
+        const history = createBrowserHistory()
+        history.replace({ search: 'q=r:golang/oauth2+test+f:travis+case:yes&patternType=regexp&c=test' })
+
+        const searchSpy = sinon.spy(defaultProps.streamSearch)
+
+        const element = mount(
+            <BrowserRouter>
+                <StreamingSearchResults
+                    {...defaultProps}
+                    history={history}
+                    location={history.location}
+                    streamSearch={searchSpy}
+                />
+            </BrowserRouter>
+        )
+
+        sinon.assert.calledOnce(searchSpy)
+        sinon.assert.calledWith(
+            searchSpy,
+            'r:golang/oauth2 test f:travis  case:yes',
+            'V2',
+            SearchPatternType.regexp,
+            'test'
+        )
+
+        element.unmount()
+    })
+
+    it('should update patternType if different between URL and context', () => {
+        const history = createBrowserHistory()
+        history.replace({ search: 'q=r:golang/oauth2+test+f:travis&patternType=regexp' })
+
+        const setPatternTypeSpy = sinon.spy()
+
+        const element = mount(
+            <BrowserRouter>
+                <StreamingSearchResults
+                    {...defaultProps}
+                    history={history}
+                    location={history.location}
+                    patternType={SearchPatternType.literal}
+                    setPatternType={setPatternTypeSpy}
+                />
+            </BrowserRouter>
+        )
+
+        sinon.assert.calledOnce(setPatternTypeSpy)
+        sinon.assert.calledWith(setPatternTypeSpy, SearchPatternType.regexp)
+
+        element.unmount()
+    })
+
+    it('should not update patternType if different between URL and context', () => {
```

Comment:
The patternType prop and the query param in the URL are both regexp. Should this be this?:
```suggestion
    it('should not update patternType if URL and context are the same', () => {
```